### PR TITLE
Add validation on product request dto and product update request dto

### LINKED ISSUE #85: Add validation on product request dto and product update request dto

### PR DESCRIPTION
**Pull request by SmrutiSnigdhaSenapati**
- [x] Add validation on product request dto and product update request dto

### CODE REVIEW

File: backend/src/main/java/com/ApparelAvenue/backend/controller/ProductController.java [hunk 1]
```diff
@@ -54,7 +55,8 @@ public ResponseEntity<Product> deleteProductById(@PathVariable String id) {
     }
 
     @PostMapping
-    public ResponseEntity<Product> save(@RequestBody ProductRequestDto dto) {
+    public ResponseEntity<Product> save(@Valid @RequestBody ProductRequestDto dto) {
+        System.out.println("Received request: " + dto);
```
Comment (smruti-banty) [inline backend/src/main/java/com/ApparelAvenue/backend/controller/ProductController.java]: Remove system.out.println

**Review by smruti-banty — CHANGES_REQUESTED** [review]
Also add this login for Product update

**Comment by SmrutiSnigdhaSenapati** [comment]
- [x] Add validation on product request dto and product update request dto

**Review by smruti-banty — APPROVED** [review]
LGTM.

**Pull request merged**